tests/test_external_connection.py

```diff
@@ -0,0 +1,120 @@
+
+    def tearDown(self):
+        pass
+
+    def test_external_connection_subscribes_to_register_unregister(self):
+        self.external_connection.pubsub.subscribe.assert_called_once_with(
+            **{
+                "base-area/register_participant":
+                    self.external_connection.channel_register_callback,
+                "base-area/unregister_participant":
+                    self.external_connection.channel_unregister_callback
+            }
+        )
+
+    def test_register_message_adds_area_to_buffer(self):
+        self.external_connection.channel_register_callback(
+            {"data": json.dumps({"name": "berlin"})}
+        )
+        assert self.external_connection.areas_to_register == ["berlin"]
+
+    def test_unregister_message_adds_area_to_buffer(self):
+        self.external_connection.channel_unregister_callback(
+            {"data": json.dumps({"name": "berlin"})}
+        )
+        assert self.external_connection.areas_to_unregister == ["berlin"]
+
+    def test_register_new_areas_creates_new_child_areas_from_buffer(self):
+        area_list = ["kreuzberg", "friedrichschain", "prenzlauer berg"]
+        self.external_connection.areas_to_register = area_list
+        self.external_connection.register_new_areas()
+        assert len(self.area.children) == 3
+        assert not self.external_connection.areas_to_register
+        assert set(ch.name for ch in self.area.children) == set(area_list)
+        assert all(isinstance(ch.strategy, self.ext_strategy_mock) for ch in self.area.children)
+        assert all(ch.parent == self.area for ch in self.area.children)
+        assert all(ch.active for ch in self.area.children)
+        assert self.external_connection.redis_db.publish.call_count == 3
+        for i, created_area in enumerate(area_list):
+            assert self.external_connection.redis_db.publish.call_args_list[i][0][0] == \
+                "base-area/register_participant/response"
+            assert self.external_connection.redis_db.publish.call_args_list[i][0][1] == \
+                json.dumps({"available_publish_channels": [
+                   f"base-area/{slugify(created_area)}/offer",
+                   f"base-area/{slugify(created_area)}/offer_delete",
+                   f"base-area/{slugify(created_area)}/offer_accept"
+                ], "available_subscribe_channels": [
+                   f"base-area/{slugify(created_area)}/offers",
+                   f"base-area/{slugify(created_area)}/offer/response",
+                   f"base-area/{slugify(created_area)}/offer_delete/response",
+                   f"base-area/{slugify(created_area)}/offer_accept/response"
+                ]})
+
+    def test_unregister_areas_deletes_children_from_area(self):
+        area_list = ["kreuzberg", "friedrichschain", "prenzlauer berg"]
+        self.external_connection.areas_to_register = area_list
+        self.external_connection.register_new_areas()
```

Comment:
Very minor, but this code is called 3 times in this class. You could refactor this into a separate function.

Reply:
You are right, I'll create a new method and call it at the start of the tests. Thanks